File: output/CodeGen/avatar/Python/Java_handraft/codeforces_30_A.java
import java.util.*;

public class codeforces_30_A {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int A = in.nextInt();
        int B = in.nextInt();
        int n = in.nextInt();
        int result = 0;
        boolean flag = false;
        for (int i = -1000; i < 1001; i++) {
            if (A * (Math.pow(i, n)) == B) {
                result = i;
                flag = true;
                break;
            }
        }
        if (flag) {
            System.out.println(result);
        } else {
            System.out.println("No solution");
        }
    }
}

File: output/CodeGen/avatar/Java/Python/codeforces_30_A.py
def main():
    sc = Scanner(sys.stdin)
    A = sc.nextint()
    B = sc.nextint()
    n = sc.nextint()
    if A == 0:
        if B == 0:
            print 1
        else:
            print "No solution"
        return
    if B % A!= 0:
        print "No solution"
        return
    B /= A
    neg = B < 0
    if neg and n % 2 == 0:
        print "No solution"
        return
    if neg:
        B = -B
    for x in xrange(0, B+1):
        if pow(x, n) == B:
            print neg and -x or x
            return
    print "No solution"